tools/wptserve/wptserve/pipes.py

```diff
@@ -273,8 +274,9 @@ def slice(request, response, start, end=None):
                 (spelled "null" in a query string) to indicate the end of
                 the file.
     """
-    content = resolve_content(response)
-    response.content = content[start:end]
+    content = resolve_content(response)[start:end]
+    response.content = content
+    response.headers.set("Content-Length", len(content))
```

Comment:
This is a separate fix, yes? Do we have any tests for this? ^_^

Also: don't we want both of the header args to be binary strings? I can't see what goes on here looking quickly? But I guess the Response side probably makes sense to do so separately.

Reply:
I didn't modify this file; the change was cherry-picked from #11769 by @Ms2ger .